src/cargo/util/toml/mod.rs

```diff
@@ -655,6 +655,19 @@ impl TomlProfile {
             }
         }
 
+        if let Some(lto) = &self.lto {
+            if let StringOrBool::String(arg) = lto {
+                if arg == "true" || arg == "false" {
```

Comment:
Try as I might I could not get `Deref` pattern matching to work here. I think because `StringOrBool` doesn't implement it, so we can't use that mechanism on its contents, only match it against `String` or `&String` etc?